apps/interactor/interactor/zeroconf.py

```diff
@@ -47,30 +50,27 @@ async def finish(self, exc_typ=None, exc=None, tb=None):
 
     async def get_interactor_service_info(self):
         return AsyncServiceInfo(
-            self.options.service_type_name,
-            f"{self.options.interactor_name}.{self.options.service_type_name}",
-            addresses=[self.options.ip_address],
+            ZEROCONF_TYPE,
+            f"{self.options.name}.{ZEROCONF_TYPE}",
+            addresses=[socket.inet_aton(self.options.ip_address)],
             port=self.port,
-            properties={
-                "friendly_name": f"Photons Interactor ({VERSION})",
-            },
+            server=f"{socket.getfqdn()}.",
+            properties={"version": VERSION, "md": "Photons Interactor"},
         )
 
 
 class ip_address_spec(sb.Spec):
     def normalise(self, meta, val):
         if val is sb.NotSpecified:
-            for interface in netifaces.interfaces():
-                addresses = netifaces.ifaddresses(interface)
-                if netifaces.AF_INET not in addresses:
-                    continue
-
-                for address in addresses[netifaces.AF_INET]:
-                    if address["addr"] != "127.0.0.1":
-                        val = address["addr"]
-
-        if val is sb.NotSpecified:
-            raise BadSpecValue("Failed to figure out an IP address to use")
+            """Get the currently active/routing IP address of the local machine."""
```

Comment:
I'm not a fan of this way of doing this. How is it better?

Reply:
This method finds the device that is used to route outound via the highest priority gateway on the box. It's the better way of reducing to a single, most likely to be operational, interface.

Alternatively, we could just list all interfaces (inclding 127.0.0.1). There are a few devices on my network that appear to do that.

Reply:
> most likely to be operational, interface.

I suppose.

My way has a better opportunity for which interface to use to be configurable though.

I've decided I don't have a strong enough opinion to care either way, this way does seem to work on my laptop *shrugs*

Reply:
We need to import socket for other things so this reduces the import count by one.

But it's also a hill upon which I'm not even prepared to get slightly inconvenienced so I'm happy to be overruled.

Let me tinker with option B: advertising all IP addresses with wanton disregard for actually connectivity.